.travis.yml

```diff
@@ -7,7 +7,7 @@ install:
   - docker build --target test --tag todo-app:test .
 script:
   - docker run todo-app:test tests
-  - docker run -e MONGO_DB_USER -e MONGO_DB_PASSWORD -e MONGO_DB_NAME -e MONGO_DB_HOSTNAME todo-app:test tests_e2e
+  - docker run -e DB_CONNECTION_STRING="$DB_CONNECTION_STRING" todo-app:test tests_e2e

```

Comment:
This should be fine... not sure why it's failing. I assume `-e DB_CONNECTION_STRING` also didn't work?
Only thing I can think of is the value is wrong.

When you set it on command line, you have to be careful to wrap it in quotes because of the ampersands in the value.

Reply:
ah it was cause i hadn't escaped characters in my connection string properly!